tests/util/util.ts

```diff
@@ -104,30 +104,40 @@ export function signJwt(
   header: { kid?: string; alg?: string; [key: string]: any },
   payload: { [key: string]: any },
   privateKey: KeyObject,
-  produceValidSignature = true
+  options?: {
+    produceValidSignature?: boolean;
+    addBogusPadding?: boolean;
+  }
 ) {
   header = {
     ...header,
     alg: "alg" in header ? header.alg : "RS256",
   };
   payload = { exp: Math.floor(Date.now() / 1000 + 100), ...payload };
+  const bogusPadding = options?.addBogusPadding ? "=" : "";
   const toSign = [
-    Buffer.from(JSON.stringify(header)).toString("base64url"),
-    Buffer.from(JSON.stringify(payload)).toString("base64url"),
+    Buffer.from(JSON.stringify(header)).toString("base64url") + bogusPadding,
+    Buffer.from(JSON.stringify(payload)).toString("base64url") + bogusPadding,
   ].join(".");
-  const digestFunction =
-    JwtSignatureAlgorithms[
-      (header.alg as keyof typeof JwtSignatureAlgorithms) ?? "RS256"
-    ];
+  const alg = (header.alg as keyof typeof JwtSignatureAlgorithms) ?? "RS256";
+  // eslint-disable-next-line security/detect-object-injection
+  const digestFunction = JwtSignatureAlgorithms[alg];
   const sign = createSign(digestFunction);
   sign.write(toSign);
   sign.end();
-  const signature = sign.sign(privateKey);
-  if (!produceValidSignature) {
-    signature[0] = ~signature[0]; // swap first byte
+  const signature = sign.sign({
+    key: privateKey,
+    dsaEncoding: "ieee-p1363", // Signature format r || s (not used for RSA)
+  });
+  if (options?.produceValidSignature === false) {
+    // swap a random byte
```

Comment:
minor comment, but to me "swap" implies taking 2 random bytes and exchanges their values, vs the "bitwise NOT" operation done here on a random byte to corrupt the value